src/wallet/spend.cpp

```diff
@@ -594,9 +594,6 @@ std::optional<SelectionResult> SelectCoins(const CWallet& wallet, CoinsResult& a
     SelectionResult preselected(pre_set_inputs.total_amount, SelectionAlgorithm::MANUAL);
     preselected.AddInputs(pre_set_inputs.coins, coin_selection_params.m_subtract_fee_outputs);
     op_selection_result->Merge(preselected);
-    if (op_selection_result->GetAlgo() == SelectionAlgorithm::MANUAL) {
-        op_selection_result->ComputeAndSetWaste(coin_selection_params.min_viable_change, coin_selection_params.m_cost_of_change, coin_selection_params.m_change_fee);
```

Comment:
nit: I'd we need to recalculating waste after adding preselected inputs because otherwise wronge waste score would be passed to USDT

Reply:
funny that I mentioned this here https://github.com/bitcoin/bitcoin/pull/25685#discussion_r938067217 (last sentence) almost 3 months ago.

Will re-check it.